### DISCUSSION TITLE
Bizzare trouble with git-credential-netrc when under exwm but not i3...

### DISCUSSION THREAD

**Discussion by jsilve24** [magit/magit#4671] [Support] [answered]
I don't use gnome but keep my git credentials in a .netrc.gpg in my home directory. I built the shell script git-credential-netrc from here: https://github.com/git/git/tree/master/contrib/credential/netrc (all this was following instructions found here: https://andrearichiardi.com/blog/posts/git-credential-netrc.html). 

The relevant section of my .gitconfig has this line: 
```
[credential]
	helper = netrc
```

All this works perfectly well (e.g., I can push and pull without trouble) both from the command line (under exwm and i3) and from magit when emacs is run under i3. But bizzarely when I run it under EXWM (which I use daily, I just use i3 for debugging), trying to pull and push through magit prompts for my username and password and shows this error log:

```
  9 git … push -v origin refs/heads/master\:refs/heads/master
Pushing to https://git.overleaf.com/61ead5cc9358ebaa96948800
git: 'credential-netrc' is not a git command. See 'git --help'.
Username for 'https://git.overleaf.com': 
```

I found this issue from a while back https://github.com/magit/magit/issues/2309 that perhaps is related? but my knowledge here is too limited to figure this out myself after an hour or two of poking around (hence asking here for some assistance). 

Also post the output of:

    M-x magit-version RET
Magit v3.3.0-251-g3cfc8458, Git 2.36.0, Emacs 28.1.50, gnu/linux

Before reporting a defect please try to reproduce it using an Emacs instance in which only Magit and its dependencies have been loaded. Other packages or your configuration should not be loaded. This makes it easier to determine whether the issue lays with Magit or something else.

**Comment by jsilve24** [comment]
Update: it is certainly not an issue with my exwm configuration. I just commented out that entire configuration and booted directly into vanilla emacs and I still get the same behavior, again, which I don't get under i3 or from the command line.

**Comment by tarsius** [comment, marked as answer]
The problem likely isn't what is in your config file but what isn't.

Check the value of `(getenv "PATH")` under both wms. Is the location of `netrc` present in one case but not the other? If so, then that's what you have to fix.

**Reply by jsilve24** [reply]
Man I feel stupid. Yes it is. Not sure how but it is. 

I also had just realized that writing out the full path to git-credential-netrc as follows solved the problem 
```
[credential]
	helper = !/home/jds6696/bin/git-credential-netrc
```

**Reply by jsilve24** [reply]
Thank you!

**Reply by tarsius** [reply]
Happens all the time :stuck_out_tongue_winking_eye:
